src/main/java/uk/gov/hmcts/reform/userprofileapi/infrastructure/controllers/UserProfileController.java

```diff
@@ -34,17 +51,19 @@
 public class UserProfileController {
 
     private UserProfileService<RequestData> userProfileService;
+    private IdamService idamService;
 
-    public UserProfileController(UserProfileService<RequestData> userProfileService) {
+    public UserProfileController(UserProfileService<RequestData> userProfileService, IdamService idamService) {
```

Comment:
Passing a service in the constructor, really??? We should be using Spring dependency-injection instead.

Reply:
Used autowired now but do not know whats the reason to why dependency creation with constructor is bad practice ?

Reply:
Ok, it's not the way we are typically doing it in prd, but I guess its not a big deal. Though we should strive to be consistent across the projects.